### PR TITLE
feature(ui-ux): onboarding screen v2

### PR DESCRIPTION
**Pull request by siradji**
#### What kind of PR is this?:

/kind

#### Which issue(s) does this PR fixes?:

Fixes #

#### Developer Checklist:

- [ ] Read your code changes at least once
- [ ] Tested on iOS/Android device (e.g, No crashes, library supported etc.)
- [ ] No console errors on web
- [ ] Tested on Light mode and Dark mode*
- [ ] Your UI implementation visually matched the rendered design*
- [ ] Unit tests*
- [ ] Added e2e tests*
- [ ] Added translations*

### CODE REVIEW

File: mobile-app/app/components/ButtonV2.tsx [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+export type ButtonFillType = 'fill' | 'outline' | 'flat'
+
+interface ButtonProps extends React.PropsWithChildren<TouchableOpacityProps> {
+  fill?: ButtonFillType
+  label?: string
+  styleProps?: string
+}
+
+export function Button (props: ButtonProps): JSX.Element {
```
Comment (kyleleow) [inline mobile-app/app/components/ButtonV2.tsx:14]: Let rename to `ButtonV2` to avoid confusion

File: mobile-app/app/components/ButtonV2.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  const buttonText = isLight ? `${fill === 'fill' ? 'text-mono-dark-v2-900' : 'text-mono-light-v2-900'} ` : `${fill === 'fill' ? 'text-mono-light-v2-900' : 'text-mono-dark-v2-900'}`
+
+  const textStyle = `${props.disabled === true ? disabledText : buttonText}`
+  return (
+    <TouchableOpacity
+      {...props}
+      style={tailwind(`${styleProps} p-3.5 rounded flex-row justify-center ${buttonStyle} ${props.disabled === true ? disabledStyle : ''}`)}
+    >
+      <>
+        <Text style={(tailwind(`${textStyle} font-bold text-center`))}>
```
Comment (kyleleow) [inline mobile-app/app/components/ButtonV2.tsx:35]: Use v2 semibold font, according to figma

File: mobile-app/app/screens/WalletNavigator/screens/OnboardingV2.tsx [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+          />
+          <Button
+            fill='flat'
+            label={translate('screens/Onboarding', 'Restore Wallet')}
+            styleProps='mx-2 mt-6 mb-12'
+            onPress={() => navigator.navigate('RestoreMnemonicWallet')}
+            testID='restore_wallet_button'
+          />
+        </ImageBackground>
+        <VersionTag />
```
Comment (kyleleow) [inline mobile-app/app/screens/WalletNavigator/screens/OnboardingV2.tsx:44]: Create a `VersionTagV2` to use v2 font

File: mobile-app/app/screens/WalletNavigator/screens/components/OnboardingCarouselV2.tsx [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+  const { isLight } = useThemeContext()
+  return (
+    <View style={tailwind('flex-1 items-center justify-center py-8 px-10')}>
+      <Image
+        source={isLight ? imageLight : image}
+        style={{ width: 200, height: 136 }}
+      />
+      <View style={tailwind('h-2/6 items-center justify-center')}>
+        <ThemedText
+          style={tailwind('text-xl font-semibold text-center mt-8')}
```
Comment (kyleleow) [inline mobile-app/app/screens/WalletNavigator/screens/components/OnboardingCarouselV2.tsx:68]: use v2 font

File: mobile-app/app/screens/WalletNavigator/screens/components/OnboardingCarouselV2.tsx [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+          dark={tailwind('text-mono-dark-v2-900')}
+          light={tailwind('text-mono-light-v2-900')}
+        >
+          {translate('screens/OnboardingCarousel', title)}
+        </ThemedText>
+        <ThemedText
+          dark={tailwind('text-mono-dark-v2-900')}
+          light={tailwind('text-mono-light-v2-900')}
+          style={tailwind('font-normal text-center mt-2 mb-8')}
```
Comment (kyleleow) [inline mobile-app/app/screens/WalletNavigator/screens/components/OnboardingCarouselV2.tsx:77]: use v2 font

File: mobile-app/app/screens/WalletNavigator/screens/components/OnboardingCarouselV2.tsx [hunk 6]
```diff
@@ -0,0 +1,110 @@
+import { Dimensions, Image, ImageSourcePropType, Platform } from 'react-native'
+import SwiperFlatList from 'react-native-swiper-flatlist'
+import ImageC from '@assets/images/onboarding/welcome-screen-a.png'
```
Comment (kyleleow) [inline mobile-app/app/screens/WalletNavigator/screens/components/OnboardingCarouselV2.tsx:3]: let be consistent on the file and variable naming, e.g.:
```suggestion
import ImageA from '@assets/images/onboarding/welcome-screen-a-dark.png'
```

File: mobile-app/app/screens/WalletNavigator/screens/components/OnboardingCarouselV2.tsx [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+    imageLight={ImageDLight}
+    key={3}
+    subtitle='Access financial opportunities with dTokens minted through decentralized vaults.'
+    title='Decentralized loans'
+  />]
+
+// Needs for it to work on web. Otherwise, it takes full window size
+const { width } = Platform.OS === 'web' ? { width: '375px' } : Dimensions.get('window')
+
+export function ImageSlide ({ image, imageLight, title, subtitle }: CarouselImage): JSX.Element {
```
Comment (kyleleow) [inline mobile-app/app/screens/WalletNavigator/screens/components/OnboardingCarouselV2.tsx:58]:
```suggestion
export function ImageSlide ({ imageDark, imageLight, title, subtitle }: CarouselImage): JSX.Element {
```

File: mobile-app/app/screens/WalletNavigator/screens/components/OnboardingCarouselV2.tsx [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+  const { isLight } = useThemeContext()
+  return (
+    <SwiperFlatList
+      autoplay
+      autoplayDelay={30}
+      autoplayLoop
+      autoplayLoopKeepAnimation
+      data={slides}
+      index={0}
+      paginationActiveColor={isLight ? '#121212' : '#F2F2F2'}
```
Comment (kyleleow) [inline mobile-app/app/screens/WalletNavigator/screens/components/OnboardingCarouselV2.tsx:96]: let's use `getColor()` to get hexcode from tailwind config, e.g.: `getColor('mono-dark-v2-100')` will return #121212

File: mobile-app/app/components/VersionTagV2.tsx [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+import { ThemedText, ThemedView } from '@components/themed'
+import { translate } from '@translations'
+
+export function VersionTagV2 (): JSX.Element {
+  return (
+    <ThemedView>
+      <ThemedText
+        dark={tailwind('text-mono-dark-v2-900')}
+        light={tailwind('text-mono-light-v2-900')}
+        style={tailwind('text-sm font-medium-v2 text-center')}
```
Comment (kyleleow) [inline mobile-app/app/components/VersionTagV2.tsx:12]: let's follow figma's font weight

File: mobile-app/app/screens/WalletNavigator/screens/components/OnboardingCarouselV2.tsx [hunk 10]
```diff
@@ -0,0 +1,110 @@
+import { Dimensions, Image, ImageSourcePropType, Platform } from 'react-native'
+import SwiperFlatList from 'react-native-swiper-flatlist'
+import ImageA from '@assets/images/onboarding/welcome-screen-c-dark.png'
```
Comment (kyleleow) [inline mobile-app/app/screens/WalletNavigator/screens/components/OnboardingCarouselV2.tsx:3]: can we match the file name too?

File: mobile-app/app/screens/WalletNavigator/screens/components/OnboardingCarouselV2.tsx [hunk 11]
```diff
@@ -0,0 +1,110 @@
...
+  const { isLight } = useThemeContext()
+  return (
+    <SwiperFlatList
+      autoplay
+      autoplayDelay={30}
+      autoplayLoop
+      autoplayLoopKeepAnimation
+      data={slides}
+      index={0}
+      paginationActiveColor={isLight ? getColor('mono-dark-v2-100') : getColor('mono-dark-v2-900')}
```
Comment (kyleleow) [inline mobile-app/app/screens/WalletNavigator/screens/components/OnboardingCarouselV2.tsx:96]: to be consistent with the design system
```suggestion
      paginationActiveColor={isLight ? getColor('mono-dark-v2-900') : getColor('mono-dark-v2-900')}
```

**Comment by fullstackninja864** [comment]
There are two different bg grid image for light and dark mode

File: mobile-app/app/screens/WalletNavigator/screens/OnboardingV2.tsx [hunk 12]
```diff
@@ -0,0 +1,48 @@
...
+          <ButtonV2
+            label={translate('screens/Onboarding', 'Get started')}
+            styleProps='m-2 mt-20'
+            onPress={() => navigator.navigate('CreateWalletGuidelines')}
+            testID='get_started_button'
+          />
+          <ButtonV2
+            fill='flat'
+            label={translate('screens/Onboarding', 'Restore Wallet')}
+            styleProps='mx-2 mt-6 mb-12'
```
[the remark this answers is not in the document]
Comment (siradji) [reply mobile-app/app/screens/WalletNavigator/screens/OnboardingV2.tsx:39]: @kyleleow ,  mine is different. 
image

File: mobile-app/app/components/VersionTagV2.tsx [hunk 13]
```diff
@@ -0,0 +1,20 @@
+import { tailwind } from '@tailwind'
+import { nativeApplicationVersion } from 'expo-application'
+import { ThemedText, ThemedView } from '@components/themed'
+import { translate } from '@translations'
+
+export function VersionTagV2 (): JSX.Element {
+  return (
+    <ThemedView>
```
Comment (kyleleow) [inline mobile-app/app/components/VersionTagV2.tsx:8]: v1 `<ThemedView>` has a different default bg color, tried replacing with `<ThemedViewV2>`
image

File: mobile-app/app/screens/WalletNavigator/screens/components/OnboardingCarouselV2.tsx [hunk 14]
```diff
@@ -0,0 +1,110 @@
...
+// Needs for it to work on web. Otherwise, it takes full window size
+const { width } = Platform.OS === 'web' ? { width: '375px' } : Dimensions.get('window')
+
+export function ImageSlide ({ imageDark, imageLight, title, subtitle }: CarouselImage): JSX.Element {
+  const { isLight } = useThemeContext()
+  return (
+    <View style={tailwind('flex-1 items-center justify-center py-8 px-10')}>
+      <Image
+        source={isLight ? imageLight : imageDark}
+        style={{ width: 220, height: 136 }}
```
Comment (kyleleow) [inline mobile-app/app/screens/WalletNavigator/screens/components/OnboardingCarouselV2.tsx:64]: let's follow the image's dimension, cropped issue is originated from image itself, design team to update the image again
```suggestion
        style={{ width: 200, height: 136 }}
```

File: mobile-app/app/screens/WalletNavigator/screens/OnboardingV2.tsx [hunk 15]
```diff
@@ -0,0 +1,63 @@
...
+        >
+          <ButtonV2
+            label={translate('screens/Onboarding', 'Get started')}
+            styleProps='m-2 mt-20'
+            onPress={() => navigator.navigate('CreateWalletGuidelines')}
+            testID='get_started_button'
+          />
+          <ButtonV2
+            fill='flat'
+            label={translate('screens/Onboarding', 'Restore Wallet')}
```
Comment (thedoublejay) [inline mobile-app/app/screens/WalletNavigator/screens/OnboardingV2.tsx:53]:
```suggestion
            label={translate('screens/Onboarding', 'Restore wallet')}
```

File: mobile-app/app/components/VersionTagV2.tsx [hunk 16]
```diff
@@ -0,0 +1,20 @@
...
+export function VersionTagV2 (): JSX.Element {
+  return (
+    <ThemedViewV2>
+      <ThemedTextV2
+        dark={tailwind('text-mono-dark-v2-900')}
+        light={tailwind('text-mono-light-v2-900')}
+        style={tailwind('text-sm font-normal-v2 text-center')}
+        testID='version_tag'
+      >
+        {translate('components/VersionTag', 'Version {{number}}', { number: nativeApplicationVersion ?? '0.0.0' })}
```
Comment (thedoublejay) [inline mobile-app/app/components/VersionTagV2.tsx:15]: Version tag is a bit too big on the first screen. Can we reduce the font size?

**Review by thedoublejay — CHANGES_REQUESTED** [review]

File: mobile-app/app/screens/WalletNavigator/screens/OnboardingV2.tsx (hunk 15 above)
Comment (kyleleow) [reply mobile-app/app/screens/WalletNavigator/screens/OnboardingV2.tsx]: updated

File: mobile-app/app/components/VersionTagV2.tsx (hunk 16 above)
Comment (kyleleow) [reply mobile-app/app/components/VersionTagV2.tsx]: updated

**Pull request merged**